### PR TITLE
feat: Faster interleaving via sequential writes, branchless column loops, and a single-block copy fast path.

### PR DESCRIPTION
**Pull request by guitarrapc**
## Summary

Optimize `BinaryInterleaver.InterleaveCodewords` (QR codeword interleaving): add an identity-copy fast path for single-block versions and replace the per-element indexed loops with a bounds-check-free transpose using upfront range validation, ref arithmetic, and additive strides. Also fixes an incorrect v5-Q example in the `CalculateInterleavedSize` doc comment (62 data codewords / 135 bytes, not 80 / 152).

## Intent

The previous implementation paid, per element: a bounds check on both source and destination, an index multiplication, and per-row-invariant `if (i < CodewordsInGroupN)` branches evaluated inside the block loop. For single-block versions (v1–v2 at most ECC levels, v5-L, ...) interleaving is the identity permutation, yet the full round-robin loop still ran.

Variant search was done with BenchmarkDotNet (15 iterations + JIT disassembly) over real ECC block patterns. Notable refuted hypotheses, so they don't get retried later:

- Inverted loop order (sequential reads / strided writes) — consistently slower than sequential writes / strided reads.
- Wide-load chunking (one 64/32-bit load per 8/4 codewords, scattered byte stores) — no reliable win; the workload is L1-resident and the scattered stores cancel the saved loads.

## Expected behavior

No functional change — output is byte-identical to the previous implementation. Verified by a new parity test (`BinaryInterleaverParityTest`) comparing against a naive reference across all 160 version/ECC-level patterns × 3 seeds, plus non-standard block-shape edge cases.

One intentional hardening: undersized `data`/`ecc`/`output` buffers now throw `ArgumentException` with a clear message upfront (previously `IndexOutOfRangeException` mid-write), guarding the ref arithmetic.

## Benchmarks

Kernel (micro-benchmark, AMD Ryzen 9 7950X3D, .NET 10, 15 iterations):

| Scenario | Blocks | Before | After | Speedup |
|---|---|---:|---:|---:|
| v1-M | 1 | 24.0 ns | 4.7 ns | **~5.1x** |
| v5-Q | 2+2 | 82.4 ns | 41.7 ns | ~2.0x |
| v40-L | 19+6 | 1,975 ns | 859 ns | ~2.3x |
| v40-H | 20+61 | 1,902 ns | 953 ns | ~2.0x |

Allocations: 0 B before and after.

End-to-end (`QrCodeEndToEnd`, coarse 1-iteration config):

| Scenario | Before | After |
|---|---:|---:|
| Short_M | 2.75 µs | 2.67 µs |
| Url_M | 6.34 µs | 6.26 µs |
| Long_L | 152.0 µs | 152.1 µs |
| Long_H | 142.1 µs | 143.0 µs |

Interleaving is ~1.3% of end-to-end generation time, so the E2E impact is small (slight gain on small QR codes, noise-level on large ones); the win is at the kernel level.

### CODE REVIEW

File: src/SkiaSharp.QrCode/Internals/BinaryEncoders/BinaryInterleaver.cs [hunk 1]
```diff
@@ -17,44 +20,103 @@ public static void InterleaveCodewords(ReadOnlySpan<byte> data, ReadOnlySpan<byt
...
+            for (var b = 0; b < totalBlocks; b++)
             {
-                var eccOffset = blockIndex * eccInfo.ECCPerBlock;
-                output[outputIndex] = ecc[eccOffset + i];
-                outputIndex++;
+                dst = Unsafe.Add(ref eccSrc, idx);
+                dst = ref Unsafe.Add(ref dst, 1);
+                idx += eccPerBlock;
             }
         }
```
[the remark this answers is not in the document]
Comment (guitarrapc) [reply src/SkiaSharp.QrCode/Internals/BinaryEncoders/BinaryInterleaver.cs:128]: Good catch on the fragile contract. Note it's not reproducible today: the only caller allocates the buffer with a plain `stackalloc` (no `SkipLocalsInit` anywhere in the repo), so the tail byte is always zero-initialized — and the previous implementation had the same caller dependency. That said, relying on the caller to hand over a pre-zeroed buffer is brittle (a future `[SkipLocalsInit]` or pooled buffer would make the remainder bits nondeterministic), so the function now clears the tail right after validation, covering both the single-block fast path and the general path. Added regression tests that pass a `0xFF`-filled buffer and assert the remainder byte is zeroed (multi-block and fast-path, with and without remainder bits).
